demo/components/fdp/fdp-service.js

```diff
@@ -111,6 +115,27 @@ app.service('FDP', function ($http, HttpEndpoint, File, Statistics, Log, $q, $ro
 	  console.log('failed uberloop');
 	});
       });
-    }
+    },  
+    getFdps: function() {
+    	return File.read('data/query/listCachedFdps.sparql').then(function(query) {
+    		return HttpEndpoint.query(query).then(function(response) {
+    			var fairDataPoints = [];    			
+    			response.data.results.bindings.forEach(function(binding) {
+    				var fdpFromCookies = $cookies.getObject(btoa(binding.fdp.value));
+    				var state = true;
+    				if (fdpFromCookies != null) {
+    					state = fdpFromCookies.state;
+    				}
+    				fairDataPoints.push({
+    					url: binding.fdp.value,
+    					name: binding.name.value,
+    					state: state 
+    				});
+    			});
+    			
+    			return fairDataPoints;
```

Comment:
Fancy work on handling the promises, who wrote that beautiful code? 😄

Reply:
Ha ha ha nice one ;-)